src/Npgsql/Internal/NpgsqlDatabaseInfo.cs

```diff
@@ -200,6 +200,16 @@ public bool TryGetPostgresTypeByName(string pgName, [NotNullWhen(true)] out Post
 
             // If the name was found but the value is null, that means that there are
             // two db types with the same name (different schemas).
+            // Try to fall back to search_path + pgName.
+            if (searchPath?.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries) is { Length: > 0 } searchPaths)
```

Comment:
That's not entirely correct as PG always searches first in pg_catalog unless it's specified, but it's probably fine?